config/config.go

```diff
@@ -31,6 +31,21 @@ type MessageProtocolConfig struct {
 	Settings      map[string]interface{} `mapstructure:"settings" json:"settings"`
 }
 
+type ManualWarpMessage struct {
+	UnsignedMessageBytes    string `mapstructure:"unsigned-message-bytes" json:"unsigned-message-bytes"`
+	SourceBlockchainID      string `mapstructure:"source-blockchain-id" json:"source-blockchain-id"`
+	DestinationBlockchainID string `mapstructure:"destination-blockchain-id" json:"destination-blockchain-id"`
+	SourceAddress           string `mapstructure:"source-address" json:"source-address"`
+	DestinationAddress      string `mapstructure:"destination-address" json:"destination-address"`
+
+	// convenience fields to access the values after initialization
+	unsignedMessageBytes    []byte
+	sourceBlockchainID      ids.ID
+	destinationBlockchainID ids.ID
+	sourceAddress           common.Address
+	destinationAddress      common.Address
+}
```

Comment:
I think that would make `Config` more difficult to use, since the programmer would need to know to _not_ use the `cfg.ManualWarpMessages` and instead use `cfg.ManualWarpMessagesParsed`. We could instead set the JSON field to be `cfg.ManualWarpMessagesUnparsed` and have the parsed field be `cfg.ManualWarpMessages`, but then the `Config` struct becomes more difficult to read.

IMO there's no obviously cleanest solution, so I'm going to keep it as is. The benefit here is that there's only a single struct to worry about, and it's never incorrect to reference the unparsed fields directly, you'd just need to parse them again at the callsite.

Reply:
Yeah, I'm not picky about this one. Was just hoping there was an cleaner way but I think you're right.